evennia/contrib/grid/extended_room/tests.py

```diff
@@ -63,7 +63,11 @@ def test_cmdextendedlook(self):
             "here",
             "Room(#{})\n{}".format(rid, self.SPRING_DESC),
         )
-        self.call(extended_room.CmdExtendedRoomLook(), "testdetail", self.DETAIL_DESC)
+        self.call(
+            extended_room.CmdExtendedRoomLook(), 
+            "testdetail", 
+            "You look closely at the {}.\n|{}".format("testdetail", self.DETAIL_DESC)
```

Comment:
Since we are now adding a third-party view to this call, consider verifying by checking what char2 sees (after making sure they are in the same room).

Reply:
Sorry, I have no idea where to start with this.  This is the first automated test I've ever created, could you point me in the right direction and I'll have a crack at it (even if it's just a similar test that I can try to push in)

Reply:
Np, since these tests inherit from `BaseEvenniaCommandTest`, there are a few basic entities available. Notably `self.char1` and `self.char2`, both of which start in the same location (found as `self.room`). 

What happens when you do `self.call(cmdname(), "argument", "expected return")` is that `self.char1` is firing the command and then we investigate which `.msg(text="")` was called on it. In this case we want to find out what `msg()` call was called on `char2` (the other person in the room by default). We will use a `mock` for this (see the Evennia documentation on unit testing and Python docs on mocks in general). Here's an example: 

```python 
# in the extended_room testing module 
from mock import Mock 
# ,,, 
class TestExtendedRoom(BaseEvenniaCommandTest): 
    # ...
    def test_looking_at_detail_with_multiple_people(self):
        # we know char2 is already in the same room, otherwise we could do something 
        # like `self.char2.location = self.char1.location` here

        # replace char2.msg with a Mock; this disables it and will catch what it is called with
        self.char2.msg = Mock() 

        # let char1 look at the detail set up for the testing room we are in 
        self.call(extended_room.CmdExtendedRoomLook(), "testdetail", self.DETAIL_DESC) 
     
        # now check what char2 saw.
        self.char2.msg.assert_called_with(text=(f"Char looks closely at testdetail."))  # not complete 

``` 
Run the test from the gamedir with `evennia test evennia.contrib.grid.extended_room.TestExtendedRoom.test_looking_at_detail_with_multiple_people`. The example above will fail because I didn't fill in the `assert_called_with` fully in my example above, but you should easly see what is sends to `msg()` and update your test assert to match.

Reply:
@Antrare I gave a suggestion on how to add the unit test above; let me know if something more is needed. Once the test is in place, this looks like it should be fine to merge :)

Reply:
Added the test, hopefully all good.